### PR TITLE
refactored query function with Flask-RESTPlus

### PR DESCRIPTION
**Pull request by liondon**

### CODE REVIEW

**Comment by nspektor** [comment]
Do you know why the code coverage decreased here?

File: service/service.py [hunk 1]
```diff
@@ -289,11 +279,14 @@ def get(self):
...
         ]
-        app.logger.info(request.args)
-        if any(i in request.args for i in filters):
-            promotions = Promotion.find_by_query_string(request.args)
+
+        args = promotion_args.parse_args()
+        app.logger.info(args)
+        if any(i in args for i in filters):
+            promotions = Promotion.find_by_query_string(args)
         else:
```
Comment (nspektor) [inline service/service.py:287]: I think this else might never get reached? which might be why the test coverage is down

**Review by nspektor — CHANGES_REQUESTED** [review]
Other than the code coverage, this looks great. When its fixed I can merge this in

File: service/service.py (hunk 1 above)
Comment (liondon) [reply service/service.py]: Thanks for pointing this out. It's now fixed. 

With the `RequestParser()`, we'll always have a dictionary with all the possible args. If the arg wasn't in the query string, the value would just be `None`. Here's what we'll get when GET `/promotions` with empty query string:

`{'title': None, 'promo_code': None, 'promo_type': None, 'amount': None, 'start_date': None, 'end_date': None, 'duration': None, 'active': None, 'is_site_wide': None, 'product': None} `

**Review by nspektor — APPROVED** [review]
good job!

**Comment by liondon** [comment]
IBM Cloud [toolchain](https://cloud.ibm.com/devops/toolchains/82ed68f3-a995-419e-80a4-f9191b98d0a4?env_id=ibm:yp:us-south "nyu-devops-promo-toolchain"): [Delivery Pipeline](https://cloud.ibm.com/devops/pipelines/a90cf2f8-b961-4e4c-98ff-49891084d12b "nyu-devops-promo-pipeline") deployed [promotions](https://github.com/2020-nyu-devops/promotions/releases/tag/deploy-ibm-yp-us-south-2020-nyu-devops-dev-20201209-195539 "deploy-ibm-yp-us-south-2020-nyu-devops-dev-20201209-195539") to [dev](https://console.bluemix.net/apps/42c5c4c9-a7f2-472a-bfb1-add0f77b8a15?env_id=ibm:yp:us-south "ibm:yp:us-south:2020-nyu-devops:dev"), including commit f3dfb8dc0e9f8d8239af0e47edf4c8d9fd7de85f

**Pull request merged**